Доработана функциональность по техническому заданию спринта

LeoVM07 opened the pull request.

LeoVM07 committed:
--Из класса InMemoryHistoryManager было удалено поле List<Task> historyOfTasks;
--Добавлен внутренний класс Node;
--Добавлены поля Node<Task> first, Node<Task> last, Map<Integer, Node<Task>> idToNode;
--Добавлен ряд методов (linkLast, remove, removeNode) для реализации двусвязного списка в классе;
--В результате история задач теперь не ограничена в размере, и в ней не будут отображаться дублирующиеся запросы, а только их последняя копия;
--В классе InMemoryTaskManager была изменена реализация методов addTask, addEpic, addSubTask для защиты задач из списка от изменений со стороны пользователя без использования менеджера задач.
--В тестовые классы был внесён ряд методов, проверяющий методы на работоспособность новых функций.

LeoVM07 committed:
--Были внесены правки в InMemoryTaskManagerTest для прохождения тестов на GitHub

LeoVM07 committed:
--Были внесены правки в InMemoryTaskManagerTest для прохождения тестов на GitHub

LeoVM07 committed:
Merge remote-tracking branch 'origin/sprint_6-solution' into sprint_6-solution

LeoVM07 committed:
Merge remote-tracking branch 'origin/sprint_6-solution' into sprint_6-solution

LeoVM07 committed:
Merge remote-tracking branch 'origin/sprint_6-solution' into sprint_6-solution

LeoVM07 committed:
--Внесены правки для прохождения тестов на GitHub

LeoVM07 committed:
Merge remote-tracking branch 'origin/sprint_6-solution' into sprint_6-solution

LeoVM07 committed:
--Внесены правки для прохождения тестов на GitHub

LeoVM07 committed:
Merge remote-tracking branch 'origin/sprint_6-solution' into sprint_6-solution

LeoVM07 committed:
--Добавлена директория Test, в которую были перенесены все необходимые тесты для работоспособности кода

LeoVM07 committed:
--Добавлена директория test, в которую были перенесены все необходимые тесты для работоспособности кода

LeoVM07 committed:
--Исправлены тестовые классы SubTaskTest и TaskTest;

litvitnik commented on src/InMemoryHistoryManager.java:
```diff
@@ -1,29 +1,84 @@
+        this.idToNode = new HashMap<>();
+    }
+
+    private static class Node<T> {
```
Приватный класс Node и все приватные методы должны находиться в конце класса)

litvitnik commented on src/InMemoryHistoryManager.java:
```diff
@@ -1,29 +1,84 @@
     }
 
+    @Override
+    public List<Task> getHistory() {
```
Также можно создать метод getTasks, который будет собирать все задачи из него в обычный ArrayList и уже его возвращать в методе getHistory. Он тоже должен быть приватным и находиться в конце класса)

LeoVM07 committed:
--Внутренний класс Node и все private методы были перенесены в конец класса InMemoryHistoryManager
--Также в нём был реализован метод getTasks, который возвращает список задач и вызывается в методе geHistory;
